### PR TITLE
[CMS-236] Add image upload endpoint

### PR DESCRIPTION
**Pull request by angary**
A PR to handle the uploading of images, though have some questions:

- Should we make a new private function to creation and adding of entities to the fs, so that `CreateNewEntity` and `UploadImage` can call it, reducing additional code
- Should we also add logging to `ParseParamsToSchema`?

### CODE REVIEW

File: Backend/endpoints/filesystem_endpoints.go [hunk 1]
```diff
@@ -203,3 +203,55 @@ func mapOver[T any, V any](input []T, mapFunction func(T) V) []V {
...
 	return output
 }
+
+type ValidImageUploadRequest struct {
+	Parent      int
+	LogicalName string `schema:"LogicalName,required"`
+	OwnerGroup  int    `schema:"OwnerGroup,required"`
+}
+
+func UploadImage(w http.ResponseWriter, r *http.Request, df DependencyFactory, log *logger.Log) (int, interface{}, error) {
```
Comment (Varun-Sethu) [inline Backend/endpoints/filesystem_endpoints.go:213]: Do we want tests? :P

File: Backend/endpoints/util.go [hunk 2]
```diff
@@ -41,6 +41,11 @@ func ParseParamsToSchema(r *http.Request, acceptingMethod string, target interfa
 		return http.StatusBadRequest
 	}
 
+	// Parse multipart file, with max size of 10 MB
+	if err = r.ParseMultipartForm(10 << 20); err != nil {
```
Comment (Varun-Sethu) [inline Backend/endpoints/util.go:45]:
```go
var maxSize uint32 = 10 << 20 
if err = r.ParseMultipartForm(maxSize); err != nil {
    // ... do stuff
}
```

File: Backend/endpoints/util.go [hunk 3]
```diff
@@ -41,6 +41,11 @@ func ParseParamsToSchema(r *http.Request, acceptingMethod string, target interfa
...
+		return http.StatusBadRequest
```
Comment (Varun-Sethu) [inline Backend/endpoints/util.go:46]: The tests are failing here. HTTP requests without a multipart file attached to them will throw an error when you try and parse them. Perhaps we should have a separate method for parsing a schema with a multipart file

File: Backend/endpoints/filesystem_endpoints.go [hunk 4]
```diff
@@ -203,3 +204,67 @@ func mapOver[T any, V any](input []T, mapFunction func(T) V) []V {
...
+		log.Write([]byte("Failed request!"))
+		return http.StatusNotAcceptable, nil, err
+	}
+
+	// Create and get a new entry in docker file system
+	dockerRepository.AddToVolume(strconv.Itoa(e.EntityID))
+	log.Write([]byte(fmt.Sprintf("Created new entity %v.", entityToCreate)))
+	dockerFile, err := dockerRepository.GetFromVolume(strconv.Itoa(e.EntityID))
+	if err != nil {
+		log.Write([]byte("Unable to get docker file"))
```
Comment (Varun-Sethu) [inline Backend/endpoints/filesystem_endpoints.go:255]: Nitpick: You dont have a fullstop here but you have one everywhere else :P

**Review by Varun-Sethu — APPROVED** [review]

File: Backend/endpoints/filesystem_endpoints.go (hunk 4 above)
Comment (angary) [reply Backend/endpoints/filesystem_endpoints.go]: Guess I'll die

**Pull request merged**